src/spartan_http_handler.erl

```diff
@@ -0,0 +1,164 @@
+-module(spartan_http_handler).
+
+-export([
+    init/3,
+    rest_init/2,
+    allowed_methods/2,
+    content_types_provided/2,
+    process/2
+]).
+
+-include_lib("dns/include/dns.hrl").
+-include_lib("erldns/include/erldns.hrl").
+
+init(_Transport, Req, Opts) ->
+    {upgrade, protocol, cowboy_rest, Req, Opts}.
+
+rest_init(Req, Opts) ->
+    {ok, Req, Opts}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>], Req, State}.
+
+content_types_provided(Req, State) ->
+    {[
+        {{<<"application">>, <<"json">>, []}, process}
+    ], Req, State}.
+
+process(Req, State = [version]) ->
+    Applications = application:which_applications(),
+    {spartan, _V, Version} = lists:keyfind(spartan, 1, Applications),
+    Body =
+        jsx:encode([
+            {<<"Service">>, <<"Spartan">>},
+            {<<"Version">>, list_to_binary(Version)}
+        ]),
+    {Body, Req, State};
+
+process(Req, State = [config]) ->
+    reply_halt(501, Req, State); % TODO
+
+process(Req, State = [hosts]) ->
+    {Host, Req0} = cowboy_req:binding(host, Req),
+    Query = #dns_query{name = Host, type = ?DNS_TYPE_A},
+    DNSMessage = #dns_message{
+        rd = true, ad = true,
+        qc = 1, questions = [Query],
+        adc = 1, additional = [#dns_optrr{}]},
+    case spartan_handler_fsm:call(DNSMessage) of
```

Comment:
If I understand properly, we're doing a DNS query to ourselves? Is there a way to just query erl-dns directly instead? Or keep around a copy of the records? It seems a bit wasteful to invoke the handler fsm logic for http queries as it will always be talking to the local node.